include/materials/ConcreteExpansionEigenstrainBase.h

```diff
@@ -0,0 +1,177 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef CONCRETEEXPANSIONEIGENSTRAINBASE_H
+#define CONCRETEEXPANSIONEIGENSTRAINBASE_H
+
+#include "ComputeEigenstrainBase.h"
+#include "RankTwoTensor.h"
+
+#include <array>
+
+class ConcreteExpansionEigenstrainBase;
+
+template <>
+InputParameters validParams<ConcreteExpansionEigenstrainBase>();
+
+/**
+ * ConcreteExpansionEigenstrainBase is a base class for computing the
+ * eigenstrain tensor to represent the effect of expansive reactions
+ * in concrete. A class that derives from this base class must define
+ * the scalar volumetric expansion. This base class applies that
+ * in either an isotropic or anisotropic manner.
+ *
+ * In the anisotropic case, the following technique is used:
+ * V. Saouma and L. Perotti. Constitutive model for alkali-aggregate
+ * reactions. ACI Materials Journal, 103(3), 2006.
+ */
+class ConcreteExpansionEigenstrainBase : public ComputeEigenstrainBase
+{
+public:
+  ConcreteExpansionEigenstrainBase(const InputParameters & parameters,
+                                   const std::string volumetric_expansion_name);
+
+  void initQpStatefulProperties() override;
+  void computeQpEigenstrain() override;
+
+protected:
+  /**
+   * Compute the total volumetric strain due to a swelling reaction for
+   * the current quadrature point.
+   * @return volumetric strain
+   */
+  virtual Real computeQpVolumetricStrain() = 0;
+
+  /**
+   * Does the derived class need the eigenvalues of the stress tensor?
+   * @return true if it does
+   */
+  virtual bool needStressEigenvalues() = 0;
+
+  /// Enum used to specify type of expansion
+  enum class ExpansionType
+  {
+    Isotropic,
+    Anisotropic
+  };
+  /// Type of expansion
+  const ExpansionType _expansion_type;
+
+  /// Compressive strength of concrete (Pa) input and stored as a positive number
+  const Real _f_compress;
+
+  /// Upper bound compressive stress beyond which no expansion occurs (in Pascals)
+  const Real _f_u;
+
+  /// Tensile strength of concrete (Pa)
+  const Real _f_tensile;
```

Comment:
What does f stand for in all of these variable names?

Reply:
f_c and f_t are commonly used to denote the compressive and tensile strength of concrete.